decidim-core/app/forms/password_validator.rb

```diff
@@ -0,0 +1,117 @@
+# frozen_string_literal: true
+
+# Class is used to verify that the user's password is strong enough
+class PasswordValidator < ActiveModel::EachValidator
+  MINIMUM_LENGTH = 10
+  MAX_LENGTH = 256
```

Comment:
On a conceptual level it seems a bit funny that a password can be "too long". But in real world, I don't think anyone would use longer passwords than this, even with proper password managers.

It seems this is a validation criteria from the `nobspw` gem, so I guess it is fine to keep as is.

It seems this defaults to the general "maximum" length of a "VARCHAR" type field in the database but this limit does not concern Decidim because the passwords are properly hashed in Devise using `BCrypt::Password` which seems to generate hashes of 60 characters.

Or maybe this is because of some general configuration limitations in web server's POST data length?

I don't know exactly but the condition to me seems a bit funny. So maybe less code, more well-being?

Reply:
BCrypt has limit of 512 chars: https://salsa.debian.org/md/libxcrypt/-/blob/master/lib/crypt.h.in#L34

Reply:
BCrypt has a limit of 512 chars: https://salsa.debian.org/md/libxcrypt/-/blob/master/lib/crypt.h.in#L34

Reply:
Yeah, so let's close this comment as resolved then.

The maximum limit is there for a reason because of the string limitation of the underlying encryption algorithm that is used to create the password hashes. It cannot create a hash if the provided string is too long (in this case 512 or longer, or whatever the underlying encryption algorithm supports, can be also shorter).

If the upper limit was not there, the user would see an exception when they post the registration form, so we need to have the upper limit to avoid this exception.

Good to know!